pkg/authorization/api/validation/validation.go

```diff
@@ -8,48 +8,126 @@ import (
 	authorizationapi "github.com/openshift/origin/pkg/authorization/api"
 )
 
-// ValidatePolicyBinding tests required fields for a PolicyBinding.
-func ValidatePolicyBinding(policyBinding *authorizationapi.PolicyBinding) fielderrors.ValidationErrorList {
+func ValidatePolicyName(name string, prefix bool) (bool, string) {
+	if name != authorizationapi.PolicyName {
+		return false, "name must be " + authorizationapi.PolicyName
+	}
+
+	return true, ""
+}
+
+func ValidatePolicy(policy *authorizationapi.Policy) fielderrors.ValidationErrorList {
 	allErrs := fielderrors.ValidationErrorList{}
+	allErrs = append(allErrs, validation.ValidateObjectMeta(&policy.ObjectMeta, true, ValidatePolicyName).Prefix("metadata")...)
+
+	for roleKey, role := range policy.Roles {
+		if roleKey != role.Name {
+			allErrs = append(allErrs, fielderrors.NewFieldInvalid("roles."+roleKey+".name", policy, "must be "+roleKey))
```

Comment:
serialized versions come in as a list of named roles, right, and we build the map in conversion? if they hit this error, does that mean we screwed up the conversion, and the error will make no sense to a user?